google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITBackupTest.java

```diff
@@ -93,6 +100,40 @@ public void setUp() throws Exception {
     instanceAdminClient = testHelper.getClient().getInstanceAdminClient();
     instance = instanceAdminClient.getInstance(testHelper.getInstanceId().getInstance());
     logger.info("Finished setup");
+
+    // Cancel any backup operation that has been started by this integration test if it has been
+    // running for at least 6 hours.
+    logger.info("Cancelling long-running test backup operations");
+    Pattern pattern = Pattern.compile(".*/backups/testbck_\\d{6}_\\d{4}_bck\\d/operations/.*");
+    try {
+      for (Operation operation :
+          dbAdminClient.listBackupOperations(instance.getId().getInstance()).iterateAll()) {
+        Matcher matcher = pattern.matcher(operation.getName());
+        if (matcher.matches()) {
+          if (!operation.getDone()) {
+            Timestamp currentTime = Timestamp.now();
+            Timestamp startTime =
+                Timestamp.fromProto(
+                    operation
+                        .getMetadata()
+                        .unpack(CreateBackupMetadata.class)
+                        .getProgress()
+                        .getStartTime());
+            long diffSeconds = currentTime.getSeconds() - startTime.getSeconds();
+            if (TimeUnit.HOURS.convert(diffSeconds, TimeUnit.SECONDS) >= 6L) {
+              logger.warning(
+                  String.format(
+                      "Cancelling test backup operation %s that was started at %s",
+                      operation.getName(), startTime.toString()));
+              dbAdminClient.cancelOperation(operation.getName());
```

Comment:
If for any reason in the future, the cancelling still doesn't work, your other option may be to just call delete on those backups. AFAIK, backups in the process of being created can be deleted.